sdk/event/event_subscription_plugin.go

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package event
+
+import (
+	"context"
+	"time"
+
+	"github.com/hashicorp/vault/sdk/helper/backoff"
+	"github.com/hashicorp/vault/sdk/logical"
+)
+
+type EventSubscriptionPlugin interface {
```

Comment:
I think the plugin interface and proto definition should probably live in `plugins/event` until we enable external plugins, otherwise it becomes part of our external facing API unnecessarily early.